### PR TITLE
[Feature] Implement `FlashAdamW` optimizer

### LINKED ISSUE #495: Support FlashAdamW
## Paper or Code

https://arxiv.org/abs/2602.23349

https://github.com/databricks/flashoptim

### PR DESCRIPTION
**Pull request by mfazrinizar**
## Problem (Why?)

Closes #495.

`pytorch-optimizer` does not yet support `FlashAdamW`, the AdamW-compatible optimizer from "FlashOptim: Optimizers for Memory-Efficient Training".

## Solution (What/How?)

Implemented `FlashAdamW` as a `BaseOptimizer` optimizer with AdamW-compatible decoupled weight decay, first and second moment state, bias correction, closure support, sparse-gradient rejection, `maximize`, and complex-parameter support through `view_as_real`.

The implementation follows the FlashOptim AdamW update semantics and adds `decouple_lr=True` for fully LR-decoupled weight decay, where the decay factor is scaled by `lr / initial_lr` instead of `lr`.

Added loader registration, package export, docs API entry, README/docs table entry, changelog note, package keyword, supported-optimizer count update, a shared optimizer training recipe, and focused tests for exact update parity, `decouple_lr`, validation, sparse gradients, closures, and loading.

## Other changes (bug fixes, small refactors)

No unrelated refactors or behavior changes.

## Notes

Nothing so far.

## Checklist

- [x] Make sure to run `make format` before commit
- [x] My code adheres to the style guidelines of this project (`make check` shows no errors)
- [x] Both new and existing unit tests pass successfully on my local environment by running `make test`
- [x] I have made the necessary changes to the documentation

### CODE REVIEW

**Review by kozistr — COMMENTED** [review]
Hi! thanks for opening the PR

after looking into this PR, it seems identical to `AdamW` optimizer, rather than FlashAdamW.

So, I think that untill we introduce the flash feature here, it'd be better not to include this yet cuz users probably confuse this

**Comment by mfazrinizar** [comment]
@kozistr Thanks for the earlier note. I updated `FlashAdamW` so it is no longer just an AdamW-compatible wrapper: it now includes FlashOptim-style grouped 8-bit optimizer-state compression, compressed/uncompressed state dict handling, and optional bf16/fp16 master-weight error-correction state. 

One question before I go further: should this PR also vendor/port the Triton fused kernels from FlashOptim, or is it acceptable to keep this as a portable PyTorch backend and leave `fused=True` unsupported for a follow-up CUDA/Triton PR? I kept `fused=True` explicit for now: it raises instead of silently pretending to use fused kernels. The full test suite and 100% coverage pass locally on CPU.

File: pytorch_optimizer/optimizer/flash_adamw.py [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+        if 'error_bits' in state:
+            state['error_bits'].copy_(_compute_ecc_bits(value, p, master_bytewidth))
+
+    def step_param(self, p: torch.Tensor, group: ParamGroup) -> None:
+        if p.grad is None:
+            return
+
+        state = self.state[p]
+        state['step'].add_(1)
+        step: int = int(state['step'].item())
```
Comment (kozistr) [inline pytorch_optimizer/optimizer/flash_adamw.py]: we could just use `group['step']` instead

**Review by kozistr — DISMISSED** [review]
thanks for your quick works!

i think it'd be better to have triton kernels in the end, but implementing the whole flash features here could be a lot and i don't have any GPU to test with for now :(

anyway, by looks of it, as-is looks good. maybe, we could add the triton kernels in a seperate PR

**Comment by mfazrinizar** [comment]
Agree. I will open a follow-up PR and work on the full Triton feature for FlashAdamW once this is merged.

File: pytorch_optimizer/optimizer/flash_adamw.py (hunk 1 above)
Comment (mfazrinizar) [reply pytorch_optimizer/optimizer/flash_adamw.py]: Updated.

**Comment by kozistr** [comment]
could you please resolve the conflicts?

**Review by kozistr — APPROVED** [review]
thanks!
